contracts/token/ERC721/extensions/ERC721Wrapper.sol

```diff
@@ -17,6 +17,10 @@ import "../utils/ERC721Holder.sol";
 abstract contract ERC721Wrapper is ERC721, ERC721Holder {
     IERC721 private immutable _underlying;
 
+    // Kept as bytes12 so it can be packed with an address
+    // Equal to 0xb125e89df18e2ceac5fd2fa8
+    bytes12 public constant WRAPPER_ACCEPT_MAGIC = bytes12(keccak256("WRAPPER_ACCEPT_MAGIC"));
```

Comment:
Can you check if Solidity is evaluating the constant at compile time or runtime?

Reply:
It's written at the bytecode but only once:

```solidity
contract ERC721Wrapped is ERC721Wrapper {
    constructor(address underlying) ERC721Wrapper(IERC721(underlying)) ERC721("Wrapped", "WRPD") {}
}
```

It appears only once in the bytecode and it always trims it:

```solidity
JUMPDEST
PUSH2 0x01fe
PUSH32 0xb125e89df18e2ceac5fd2fa8b7887eec71a2a2c0a2378e293dccc6be1aef7449
DUP2
JUMP
JUMPDEST
PUSH1 0x40
MLOAD
PUSH1 0x01
PUSH1 0x01
PUSH1 0xa0
SHL
SUB
NOT
SWAP1
SWAP2
AND
DUP2
```

One place reaching this code is the `onERC721Received` function, which selector is located exactly before the selector of the second case `WRAPPER_ACCEPT_MAGIC`

Every other instance is hardcoded like (idk why Solidity does this lmao):

```solidity
PUSH12 0x1624bd13be31c59d58bfa5f5
PUSH1 0xa3
SHL
```